### PR TITLE
SSO State Migration

### PR DESCRIPTION
**Pull request by iambmelt**
See description here: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/436

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/internal/cache/migration/DefaultTokenCacheItemAdapter.java [hunk 1]
```diff
@@ -0,0 +1,230 @@
+package com.microsoft.identity.common.internal.cache.migration;
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/migration/DefaultTokenCacheItemAdapter.java]: Delete this class

File: common/src/main/java/com/microsoft/identity/common/internal/cache/migration/ITokenCacheItemAdapter.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/cache/migration/ITokenCacheItemAdapter.java]: Delete this class

File: common/src/main/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapter.java [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+     */
+    private MicrosoftAccount createAccount(@NonNull final ADALTokenCacheItem refreshToken) {
+        final String methodName = ":createAccount";
+        try {
+            final String rawIdToken = refreshToken.getRawIdToken();
+            final String uid = refreshToken.getUserInfo().getUserId();
+            final String utid = refreshToken.getTenantId();
+            final String environment = new URL(refreshToken.getAuthority()).getHost();
+
+            JsonObject clientInfo = new JsonObject();
```
Comment (addev-ashish) [inline common/src/main/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapter.java]: nit: can be final

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapter.java]: Thanks, resolved in d609ffdd0cb995eb7812e1a5370cd2bc8482f12e

**Pull request merged**